### ISSUE TITLE
Circular Dependencies Error when there are none

### ISSUE THREAD

**Issue by nahtnam** [fabien0102/ts-to-zod#70]
## Bug description

I'm trying to use `ts-to-zod` in a file, but I'm getting an error saying that there are circular dependencies when I'm sure there are none.

### Input

Here is the main file. NOTE I changed some text around to make more sense and be less niche, so if you see any typos, that's me and not the code :)

```ts
import { DB } from '../../../components/dashboards/__generated__/DB.generated';

export type DBEntity = {
  id: string;
  cache: {
   data: DB['data'];
  } | null;
};
```

db.generated (imports only)
```
import * as Types from '../../../../../__generated__/baseTypes';
import gql from 'graphql-tag';
import { DBFragment } from './DBFragment.generated';
import * as Urql from 'urql';
```

db.fragment (imports only)
```
import * as Types from '../../../../../__generated__/baseTypes';
```

baseTypes has NO imports (only exports).

**Versions**

- Typescript: `v4.5.5`
- Zod: `v3.11.6`

**Comment by fabien0102** [comment]
Yeah… the error message is indeed missleading for this case. The actual problem is that ts-to-zod don't support multiple files at the moment.

Knowing this
* `DB['data']` can't be resolved
* so stuck in this loop -> https://github.com/fabien0102/ts-to-zod/blob/e349c33de997505da77103555f8d67446983b9f1/src/core/generate.ts#L103-L130 
* Since the lib assume you are working with single file, this must be a circular dependency! 😅 

Anyway, having multiple files support will be amazing (just lack of time, since this is not a simple task), but I can at least improve the error message in the midtime.

Thanks for using this library and the bug report. Always appreciated to have nice bug descriptions 😃
